### ISSUE TITLE
Inline editor doesn't scroll with you

### ISSUE THREAD

**Issue by triplejumper12** [tinymce/tinymce#4153]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
When you scroll down the page with the editor toolbar open, the toolbar scrolls off the top of the screen

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tinymce.com](http://fiddle.tinymce.com/) or similar.**
http://fiddle.tinymce.com/dDeaab/7 Just load it, and the toolbar should be hidden under the element above it

**What is the expected behavior?**
It should show under the element above it, and fix itself to the top of the page instead of scrolling past.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
It was working in 4.7.2, but broke starting in 4.7.3 and it is still broken in nightly and edge. I've tested Chrome and Firefox on Linux, and we've had reports that it is broken in Safari too.

**Comment by triplejumper12** [comment]
This is still broken in 4.7.5 and edge. In my fiddle example, it renders underneath the code box and you can't even use the toolbar. Just curious if this is an actual issue? Or if using the fixed_toolbar_container option and doing our own scrolling/positioning is the supported solution?
![inlineeditorissue](https://user-images.githubusercontent.com/727610/35300236-7d655a54-004d-11e8-9af4-821d18b4eb09.png)

**Comment by roelvanduijnhoven** [comment]
We experience the exact same bug!

I think it is related to these commits. Although there is no mention of any fix or change in the changelog for 2.7.3. 

* https://github.com/tinymce/tinymce/commit/387fa3b04d9cef05e5ad60e0f985f0f6afb8b26e 

  > TINY-1375: fixed so inline toolbar gets rendered inside the ui_container
* https://github.com/tinymce/tinymce/commit/4e68e6e96cb927261eb8e5ddc08f9fef55726e7c 

  > TINY-1375: fixed contextmenu position if ui container was non static
* https://github.com/tinymce/tinymce/commit/84f81cf967a1b4ad44eccc759f69e4d8e9e9fe7c 

  > TINY-1375: fixed contextmenu position if ui container was non static

I have no clue what the TINY-1375 references to. Do you have any clue to what bug tracker that is @triplejumper12?

Did you have any luck finding a solution @triplejumper12?

**Comment by triplejumper12** [comment]
@roelvanduijnhoven, I ended up using the [fixed_toolbar_container](https://www.tinymce.com/docs/configure/editor-appearance/#fixed_toolbar_container) option to throw it in a container so I could handle my own positioning with css/js.

**Comment by roelvanduijnhoven** [comment]
Haha this brings back memories from long time ago. When we initially tried to implement the floating logic we first went with our own toolbar. But had problems with the menu's not properly positioning.

The container is absolutely positioned; so it floats above the element.

I just tried the same thing; but it is still flawed. See
![screenshot from 2018-04-06 09-40-35](https://user-images.githubusercontent.com/91910/38408638-b1e31ce8-397e-11e8-9d54-bb64d65eca38.png)

Do you by any change have an idea what could be going wrong; or how you fixed this?

**Comment by roelvanduijnhoven** [comment]
We opened this StackOverflow issue in 2014 that is still releveant today :p. https://stackoverflow.com/questions/27398639/tinymce-submenus-not-sticking-to-toolbar-when-using-fixed-toolbar-container-and

**Comment by roelvanduijnhoven** [comment]
@triplejumper12 Thanks for your input.

However I should not be hijacking this issue. The issue you reported is that the floating mechanism of the inline editor is broken since 2.7.3 and is reproduced in this fiddle: http://fiddle.tinymce.com/dDeaab/7.

**Comment by villermen** [comment]
Positioning of the toolbar for inline editors is still broken across platforms almost half a year later in v2.8.3.

It is reproduced with vanilla TinyMCE even without plugins or a custom toolbar in place by http://fiddle.tinymce.com/B0gaab.

```html
<div class="editable">
  <p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p><p>text</p>
</div>

<script>
tinymce.init({
  selector: ".editable",
  inline: true,
});
</script>
```

I am running into bugs that can easily be fixed by upgrading TinyMCE. This bug is the only thing preventing me from upgrading beyond v4.7.2 in any way as it fundamentally breaks inline mode.

Any word on this would be greatly appreciated @spocke @fyrkant.

**Comment by fyrkant** [comment]
This bug/issue has been fixed in the latest version. If you can still reproduce this on the latest version please let us know and we re-open this ticket. Thanks.

**Comment by triplejumper12** [comment]
This appears to be even more broken in the v5.0.0. We were able to use the fixed_toolbar_container option, but now that it no longer exists in 5, we were forced back into the floating bar.

If you have content at the way top of your page, the bar always floats over it, so you can't see what you are editing. Also, if you have a floating menu that is fixed to the top of the page, the editor gets hidden under it completely. 
[http://fiddle.tinymce.com/dDeaab/17](http://fiddle.tinymce.com/dDeaab/17)

Is it possible to add fixed_toolbar_container back in? It made it so much easier for all of our pages using it so it is always in the same position on the page, and you don't have to look around to see where it popped up. Otherwise maybe a minimum top position so it can't get any higher than 150px from the top or something along those lines?
Thanks

**Comment by TheSpyder** [comment]
The originally reported issue has long since been fixed, making sure the editor UI remains on screen when scrolling the page.

While implementing the new UI we didn't include `fixed_toolbar_container`. We have now re-implemented it and it should be released in 5.0.2 - #4598 is the tracking number for that issue.
